src/domain/events/EventsPage.tsx

```diff
@@ -2,43 +2,87 @@ import { Koros } from 'hds-react';
 import { useTranslation } from 'next-i18next';
 import Head from 'next/head';
 import Link from 'next/link';
+import { useRouter } from 'next/router';
 import React, { ReactElement } from 'react';
 
 import LoadingSpinner from '../../common/components/loadingSpinner/LoadingSpinner';
-import { usePopularKeywordsQuery } from '../../generated/graphql';
+import {
+  usePopularKeywordsQuery,
+  useUpcomingEventsQuery,
+} from '../../generated/graphql';
 import useLocale from '../../hooks/useLocale';
 import localizedString from '../../utils/getLocalisedString';
+import getPageNumberFromUrl from '../../utils/getPageNumberFromUrl';
 import Container from '../app/layout/Container';
 import PageWrapper from '../app/layout/PageWrapper';
 import { ROUTES } from '../app/routes/constants';
 import BannerHero from '../bannerHero/BannerHero';
 import EventList from './eventList/EventList';
-import EventSearchForm, { PanelState } from './eventSearchForm/EventSearchForm';
+import EventSearchForm, {
+  EventSearchFormValues,
+  PanelState,
+} from './eventSearchForm/EventSearchForm';
 import styles from './eventsPage.module.scss';
-import { useEventsSearch } from './EventsSearchPage';
+import { getSearchQueryObject } from './utils';
 
 const panelStates = {
   closed: PanelState.Condensed,
   open: PanelState.Advanced,
 };
 
 const EventsPage = (): ReactElement => {
+  const router = useRouter();
+  const [isLoadingMore, setIsLoadingMore] = React.useState(false);
   const [searchPanelState, setSearchPanelState] = React.useState<PanelState>(
     panelStates.closed
   );
 
   const {
-    initialValues,
+    data: eventsData,
     loading,
-    events,
-    eventsCount,
-    isLoadingMore,
-    nextPage,
-    sort,
-    fetchMoreEvents,
-    search,
-    setSort,
-  } = useEventsSearch();
+    fetchMore,
+  } = useUpcomingEventsQuery({
+    variables: {
+      include: ['keywords', 'location', 'audience'],
+    },
+  });
+  const events = eventsData?.upcomingEvents?.data;
+  const eventsCount = eventsData?.upcomingEvents?.meta.count;
+
+  const search = (values: EventSearchFormValues) => {
+    values = { ...values, organisation: values.organisationId };
+    delete values.organisationId;
+
+    router.push(
+      {
+        pathname: ROUTES.EVENTS_SEARCH,
+        query: getSearchQueryObject(values),
+      },
+      undefined,
+      { scroll: false }
+    );
+  };
+
+  const nextPage = React.useMemo(() => {
+    const nextUrl = eventsData?.upcomingEvents?.meta.next;
+    return nextUrl ? getPageNumberFromUrl(nextUrl) : null;
+  }, [eventsData?.upcomingEvents?.meta?.next]);
+
+  const fetchMoreEvents = async () => {
+    if (nextPage) {
+      try {
+        setIsLoadingMore(true);
+        await fetchMore({
+          variables: {
+            page: nextPage,
+          },
+        });
+        setIsLoadingMore(false);
+      } catch (e) {
+        setIsLoadingMore(false);
+      }
+    }
+  };
```

Comment:
I don't think the upcoming events will ever have a pagination, because if paginated in Kultus, The pagination from LinkedEvents would be a problem. I may be wrong, but I thought the API implementation is so that the next is always `null`. Sure it does not wreck anything and UI doesn't need to know such a detail 😄 

How is it @charn?